src/status_im2/contexts/communities/actions/request_to_join/view.cljs

```diff
@@ -3,31 +3,35 @@
             [react-native.core :as rn]
             [react-native.gesture :as gesture]
             [reagent.core :as reagent]
+            [status-im2.common.password-authentication.view :as password-authentication]
             [status-im2.contexts.communities.actions.community-rules-list.view :as community-rules]
             [status-im2.contexts.communities.actions.request-to-join.style :as style]
             [utils.i18n :as i18n]
-            [utils.re-frame :as rf]
-            [utils.requests :as requests]))
+            [utils.re-frame :as rf]))
 
 (defn request-to-join-text
   [open?]
   (if open?
     (i18n/label :t/join-open-community)
     (i18n/label :t/request-to-join)))
 
+(defn join-community-and-navigate-back!
```

Comment:
just wondering why it has `!` mark in the end

Reply:
habit, I use it for functions that have side effects, clojure recommends it for state changing functions (`reset!` etc), but not sure we do that throughout the codebase, so I'll change it